via/services/youtube.py

```diff
@@ -11,6 +11,19 @@ def __init__(self, enabled: bool):
     def enabled(self):
         return self._enabled
 
+    def canonical_video_url(self, video_id: str) -> str:
+        """
+        Return the canonical URL for a YouTube video.
+
+        This is used as the URL which YouTube transcript annotations are
+        associated with.
+        """
```

Comment:
Is this necessary?

I understand that `view_video.html.jinja2` needs to have the `<link rel="canonical" href="{{ video_url }}"/>` but I think you could just pass the template the non-normalized URL, in `view_video.py::youtube()`@

```diff
diff --git a/via/views/view_video.py b/via/views/view_video.py
index 4c0e4b0..48770da 100644
--- a/via/views/view_video.py
+++ b/via/views/view_video.py
@@ -31,6 +31,7 @@ def youtube(request, url, **kwargs):
         "client_embed_url": request.registry.settings["client_embed_url"],
         "client_config": client_config,
         "video_id": video_id,
+        "video_url": url,
         "api": {
             "transcript": {
                 "doc": "Get the transcript of the current video",
```

I don't think it's Via's job to normalize / canonicalize YouTube URLs? Surely that's the job of the URL normalization and document equivalence code in h? Via isn't the only way that people might annotate YouTube videos and pages, after all.

Reply:
> I don't think it's Via's job to normalize / canonicalize YouTube URLs?

For web pages, we do allow and encourage web page authors to indicate what the preferred version of a URL is using `<link rel=canonical>`, which originated as a hint for search engines, and YouTube does set this themselves. They also redirect the various URLs above to this canonical URL.

> Via isn't the only way that people might annotate YouTube videos and pages, after all.

When a user visits YouTube directly, YouTube includes this same `<link rel=canonical>` link on their pages.